### ISSUE TITLE
Add Catel as a framework option

### ISSUE THREAD

**Issue by GeertvanHorrik** [Microsoft/WindowsTemplateStudio#911]
I would like to add in the [Catel](http://www.catelproject.com) framework.

I can create a local branch in the fork for WTS and create a PR to add support for this framework.

**Comment by advapiIT** [comment]
I like it

**Comment by advapiIT** [comment]
I like it!

**Comment by mrlacey** [comment]
@GeertvanHorrik before you do anything it'd be good to get a thorough understanding of what you think support for the Catel framework will look like and how it would fit in with what exists already.  
Every new framework that is supported adds an extra level of complexity, to the tool and templates, that needs to be maintained going forward so it's necessary to be very careful about adding new frameworks.

**Comment by GeertvanHorrik** [comment]
Thanks for your reply @mrlacey . I've studied the template structure and am convinced it would easily fit in where the others are. A few things where Catel differs a bit from the other frameworks is:

- Base class for the views (so the template would post action the view base class (xaml only))
- ViewModelBase is nearly identical to other frameworks (although I would recommend adding InitializeAsync and CloseAsync for convenience)
- NuGet packages installations

The rest of the templates should be compatible with Catel out of the box (except that each view in Catel automatically has their own VM). Since they are resolved by naming convention, it will work out of the box.

So basically I want to ease 2 things:
1. Project creation for Catel
2. View creation for Catel

The rest of the templates should stay as they are (or are already compatible).

**Comment by crutkas** [comment]
@GeertvanHorrik I think right now, too many other items need to happen before this could be incorporated.  Currently, we're targeting top frameworks.  PRISM, Template 10 and Caliburn are the next ones we plan on adding.  These are based off community, non-creator feedback along with looking at store data to validate priority of work.

Before we add on non-top frameworks, I think there are features as well we need to add into WTS.  We haven't really prioritized the work as we haven't had the issue

- #282 which is adding search into wts
- #14 top versus all features

One top of that, quickly, I can see that we'll need to: figure out a better ways to do unit tests to scale for these scenarios, improve unit test speeds, better way to easily support tons of frameworks with templates as (The testing permutations for adjusting #350 took me hours and i still missed one due to a false assumption of pivot vs Navigation View).  

Throw VB in the mix, everything doubles, for both unit tests and for time to test

**Comment by GeertvanHorrik** [comment]
Thanks for taking the time to write a reply. I understand the complexity you have to deal with, and respect your decision.

I have considered working on these issues you mentioned (#282 / #14), but to be honest it's out of scope for what I was trying to achieve (if there is no guarantee that eventually Catel would be accepted as fx).

**Comment by crutkas** [comment]
Another idea for this would be having multiple feeds in.  This would 100% support it then.

**Comment by GeertvanHorrik** [comment]
True. From what I understand (or at least how I see it), WTS is the replacement of item / project templates for UWP apps. Happy to work on this once all the other frameworks are supported and / or multiple feeds are an option. Until that time I'll add the item / project templates manually to the Visual Studio gallery.

**Comment by mrlacey** [comment]
Not a replacement, an alternative. 
I can't see a time when the need for individual templates are not needed. The difference is the output from a standard template is fixed (and there are lots of scenarios where people will want and need that) while the output from WTS is configurable.
If you just need something fixed then don't go to the trouble of making it configurable.

Yes, once multiple hosted feeds ( #234 ) is supported (sometime later this year) then anyone can host extra templates that can be merged with the default templates WTS includes.

**Comment by GeertvanHorrik** [comment]
> Not a replacement, an alternative.

Technically correct, but WTS is being marketed, where the item / project templates are not. Meaning new users will *always* use WTS, meaning they can't choose Catel.

Anyway, I'll keep an eye on the progress and re-evaluate later this year.

Thanks!

**Comment by crutkas** [comment]
@GeertvanHorrik we're still a feisty startup even internally at Microsoft and we have great community support.  I'm all for having it but right now we're fighting things like making our unit tests run faster and figuring out some core fundamentals to be able to support scenarios like this.

Remember, we only shipped at Build so we're 3 months old.  We're working to do the right thing for everyone.

**Comment by GeertvanHorrik** [comment]
Fully understand, no worries. Take your time ;-)
